app/services/email_settings_validator.rb

```diff
@@ -0,0 +1,177 @@
+# frozen_string_literal: true
+
+require 'net/imap'
+require 'net/smtp'
+require 'net/pop'
+
+# Usage:
+#
+# begin
+#   EmailSettingsValidator.validate_imap(host: "imap.test.com", port: 999, username: "test@test.com", password: "password")
+#
+#   # or for specific host preset
+#   EmailSettingsValidator.validate_imap(**{ username: "test@gmail.com", password: "test" }.merge(Email.gmail_imap_settings))
+#
+# rescue => err
```

Comment:
What do you think about allowlisting potential exceptions? Something like
```
class EmailSettingsValidator
  FRIENDLY_EXCEPTIONS = [Net::POPAuthenticationError, Net::IMAP::NoResponseError...]
```
And then rescue can look like
```
rescue *FRIENDLY_EXCEPTIONS => err
```

Not sure if that would work, but it would be nice to not have blank rescue

Reply:
Ah yes that makes sense, will add that